### PR TITLE
Update contacts App with PATCH behavior implementation.

### PR DESCRIPTION
**Pull request by askurydzin**
This fix could be suspended till https://github.com/grpc-ecosystem/grpc-gateway/pull/671 merged in grpc-gateway.

### CODE REVIEW

**Comment by askurydzin** [comment]
the build is failing because i used Daniel branch to generate and build sources. When Daniel's patch will be merged I'll update dep to use the latest grpc-gateway.

**Comment by askurydzin** [comment]
So, now everything seems to work well

**Comment by dkukharau** [comment]
IMHO we need to think about how to handle this field in a bit more elegant way.

**Comment by askurydzin** [comment]
For now I see no other way, if we want to employ patch behavior, what should happen when we send patch for primary_email?

a) all emails should be removed and new email (primary one) should be set

b) previous primary email should be overwritten as ordinary email

c) previous primary email should be removed

I've chosen c) option, and the only way to do it is to walk through all emails and remove one that is primary in db.

File: pkg/pb/contacts.override.gorm.go [hunk 1]
```diff
@@ -11,17 +12,48 @@ import (
...
+				*e.IsPrimary = false
+				emails = append(emails, e)
+			} else {
+				e.IsPrimary = new(bool)
+				*e.IsPrimary = true
+
+				primary = e
 			}
+		} else {
+			emails = append(emails, e)
```
[the remark this answers is not in the document]
Comment (askurydzin) [reply pkg/pb/contacts.override.gorm.go]: did it in the same way as ToORM code does.

File: pkg/pb/contacts.override.gorm.go [hunk 2]
```diff
@@ -11,17 +12,42 @@ import (
...
+		} else {
+			e.IsPrimary = new(bool)
+			*e.IsPrimary = true
+			primary = e
+		}
+	}
+
+	if primary == nil {
+		accountID, err := auth.GetAccountID(ctx, nil)
```
Comment (dkukharau) [inline pkg/pb/contacts.override.gorm.go]: I think it's not very good that app code deals with `AccountID` directly, could you create a PB Email, convert it to ORM and then set `IsPrimary` to `true`?

**Pull request merged**